### DISCUSSION TITLE
Remove/disable the "live" button when using a Blob as an audio source

### DISCUSSION THREAD

**Discussion by TasseDeCafe** [vidstack/player#1236] [Q&A] [answered]
Hi!

I'm generating an audio blob recorded with the microphone in the browser, and storing it in my src state like this:

My environment: localhost on Chrome desktop. I'm developing a React app.

```ts
const url = URL.createObjectURL(blob)
    setSrc({ src: url, type: 'audio/mpeg' })
```

In then simply pass the src state to the MediaPlayer like this:

```ts
<MediaPlayer title={title} src={src} ref={ref} viewType='audio'>
```

This works correctly, but when using the default layout, this "live" button appears:

![CleanShot 2024-04-06 at 19 22 07@2x](https://github.com/vidstack/player/assets/44320355/e02338a2-51ee-433e-8697-687073f34fd7)

I tried to disable it by passing a "null" slot, but it doesn't seem to affect it. Passing streamType='on-demand' also doesn't seem to affect it.

What am I doing wrong here? This audio source is not live, it's a pre-recorded audio blob.

Thank you for your help.

**Comment by TasseDeCafe** [comment]
As a follow-up to this, I ended up finding out that you can pass duration={duration} in the MediaPlayer, and you can calculate the of the blob like this:

```js
export const getBlobDuration = async (blob: Blob): Promise<number> => {
  const audioContext = new AudioContext()
  const arrayBuffer = await blob.arrayBuffer()
  const audioBuffer = await audioContext.decodeAudioData(arrayBuffer)
  const duration = audioBuffer.duration
  audioContext.close()
  return duration
}
```

I'm not sure if that's the best solution, but that removes the "live" button because the duration is now set.

**Reply by mihar-22** [reply]
Thanks heaps for the update @TasseDeCafe! If you can make a [repro](https://github.com/vidstack/player/blob/main/.github/REPRO.md) then I'd love to look into it further and fix any issues.

**Reply by TasseDeCafe** [reply, marked as answer]
I ended up realizing that the issue was the fact that Chrome doesn't provide the duration data for the webm format when you record audio inside the browser. The audio is also not seekable. I fixed the problem with this library: https://www.npmjs.com/package/webm-duration-fix

**Reply by lchMemory** [reply]
I'm facing the same problem. When I assign the URL of a Blob file, which is generated from browser recording, to the player, I've specified the format with streamType='on-demand', yet it still defaults to the live layout mode. I hope the player can address this issue internally. Thanks for your assistance.

![image](https://github.com/user-attachments/assets/fe79981f-08bf-408a-b1ac-b531da1c0b02)
![image](https://github.com/user-attachments/assets/a4f44b5a-aa0d-4ad0-800a-05d521ced6a0)
